### PR TITLE
Report array and map content

### PR DESCRIPTION
**Pull request by AirQuick**
This pull request derives from #545. So needs to be handled after that.

---

This pull request enables the result reports to describe `array(*)` and `map(*)`. (Only on XSLT. Not on XQuery.)
The report is made in a string. No structured diff.

#### Example of Report HTML
* [XSLT](https://htmlpreview.github.io/?https://github.com/xspec/xspec/blob/b40109118fe379b68904dd23b542ea276c63bf4d/test/end-to-end/cases/expected/stylesheet/xspec-report-result.html#ELEM-54)
* [XQuery](https://htmlpreview.github.io/?https://github.com/xspec/xspec/blob/b40109118fe379b68904dd23b542ea276c63bf4d/test/end-to-end/cases/expected/query/xspec-report-result.html#ELEM-55)

### CODE REVIEW

File: test/end-to-end/cases/xspec-report.xspec [hunk 1]
```diff
@@ -4,11 +4,25 @@
...
-			<x:param as="map(*)" select="map { 'foo': 1 }" />
-		</x:call>
-		<x:expect label="[Result] should be reported" />
+		<!-- function(*) (excluding array(*) and map(*)) requires Saxon-PE or EE -->
+
+		<x:scenario label="Array">
+			<x:call function="exactly-one">
+				<x:param as="array(*)" select="['foo', 1, [2, 'bar']]" />
+			</x:call>
+			<x:expect label="[Result] should be reported" />
```
Comment (galtm) [inline test/end-to-end/cases/xspec-report.xspec]: Would you consider a more descriptive label, such as "Serialized array (XSLT, assuming Saxon-PE/EE) or empty 'pseudo-other' element (XQuery) should be reported upon failure"?

I know this label predates this pull request, so feel free to disregard this comment or recommend a new enhancement for it.

Comment (AirQuick) [reply test/end-to-end/cases/xspec-report.xspec]: Good point. I updated this PR based on your suggestion: 85d0b245dd04bf6d1b89227c2d3ebf49361584ca

Comment (galtm) [reply test/end-to-end/cases/xspec-report.xspec]: Thanks!

**Comment by AirQuick** [comment]
@cirulls 
Merged this enhancement.

Thanks @galtm for the review.

**Pull request merged**